subsys/net/lib/lwm2m/lwm2m_rw_oma_tlv.c

```diff
@@ -961,23 +1006,41 @@ int do_write_op_tlv(struct lwm2m_message *msg)
 
 			while (pos < tlv.length &&
 			       (len2 = oma_tlv_get(&tlv2, &msg->in, true))) {
-				if (tlv2.type != OMA_TLV_TYPE_RESOURCE) {
-					pos += len2;
-					continue;
-				}
-
-				ret = write_tlv_resource(msg, &tlv2);
-				if (ret) {
-					return ret;
+				if (tlv2.type == OMA_TLV_TYPE_RESOURCE) {
+					ret = write_tlv_resource(msg, &tlv2);
+					if (ret) {
+						return ret;
+					}
+				} else if (tlv2.type == OMA_TLV_TYPE_MULTI_RESOURCE) {
+					oma_tlv_get(&tlv2, &msg->in, false);
+					ret = lwm2m_multi_resource_tlv_parse(msg, &tlv2);
+					if (ret) {
+						return ret;
+					}
+				} else {
+					/* Skip Unsupported TLV type */
+					return -ENOTSUP;
 				}
 
 				pos += len2;
 			}
 		} else if (tlv.type == OMA_TLV_TYPE_RESOURCE) {
+			if (msg->path.level < 2) {
+				return -ENOTSUP;
+			}
 			ret = write_tlv_resource(msg, &tlv);
 			if (ret) {
 				return ret;
 			}
+		} else if (tlv.type == OMA_TLV_TYPE_MULTI_RESOURCE) {
+			if (msg->path.level < 2) {
```

Comment:
And here (`LWM2M_PATH_LEVEL_OBJECT_INST`).

Reply:
Ah good point I update.